flowey/flowey_core/src/pipeline.rs

```diff
@@ -1273,18 +1324,43 @@ pub mod internal {
     #[derive(Debug, Clone)]
     pub enum Parameter {
         Bool {
+            name: String,
             description: String,
+            kind: ParameterKind,
             default: Option<bool>,
         },
         String {
+            name: String,
             description: String,
             default: Option<String>,
+            kind: ParameterKind,
             possible_values: Option<Vec<String>>,
         },
         Num {
+            name: String,
             description: String,
             default: Option<i64>,
+            kind: ParameterKind,
             possible_values: Option<Vec<i64>>,
         },
     }
+
+    impl Parameter {
+        pub fn name(&self) -> String {
+            match self {
+                Parameter::Bool { name, kind, .. } => match kind {
+                    ParameterKind::Unstable => format!("__unstable_{name}"),
+                    ParameterKind::Stable => name.to_string(),
+                },
+                Parameter::String { name, kind, .. } => match kind {
+                    ParameterKind::Unstable => format!("__unstable_{name}"),
+                    ParameterKind::Stable => name.to_string(),
+                },
+                Parameter::Num { name, kind, .. } => match kind {
+                    ParameterKind::Unstable => format!("__unstable_{name}"),
+                    ParameterKind::Stable => name.to_string(),
+                },
+            }
```

Comment:
```suggestion
            let (name, kind) = match self {
                Parameter::Bool { name, kind, .. } => (name, kind),
                Parameter::String { name, kind, .. } => (name, kind),,
                Parameter::Num { name, kind, .. } => (name, kind),
            };
            
            match kind {
                ParameterKind::Unstable => format!("__unstable_{name}"),
                ParameterKind::Stable => name.to_string(),
            }
```

Reply:
Nice, that's much better. Thank you!

Reply:
note that my other comment below likely invalidated this improvement 😅